### PR TITLE
Add `--setup` to `viash ns test` and `viash test`

### PR DESCRIPTION
**Pull request by DriesSchaumont**
## Describe your changes

Add `--setup` to `viash ns test` and `viash test`

## Issue ticket number and link
~Closes #xxxx (Replace xxxx with the GitHub issue number)~

## Checklist before requesting a review
- [x] I have performed a self-review of my code

- Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Bug fixes

- [x] Proposed changes are described in the CHANGELOG.md

- [x] Relevant unit tests have been added

### CODE REVIEW

File: .github/workflows/sbt_test.yml [hunk 1]
```diff
@@ -17,6 +17,22 @@ jobs:
...
+        sudo install -m 0755 -d /etc/apt/keyrings
+        curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+        sudo chmod a+r /etc/apt/keyrings/docker.gpg
+        echo "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+             "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
+        sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+        sudo apt-get update
+        sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+        sudo docker run hello-world
+
```
Comment (rcannood) [inline .github/workflows/sbt_test.yml:35]: Can you store this in a separate action in viash-actions?

File: src/main/scala/io/viash/cli/CLIConf.scala [hunk 2]
```diff
@@ -214,8 +214,15 @@ class CLIConf(arguments: Seq[String]) extends ScallopConf(arguments) {
     banner(
       "viash test",
       "Test the component using the tests defined in the viash config file.",
-      "viash test config.vsh.yaml [-p docker] [-k true/false]")
-
+      "viash test config.vsh.yaml [-p docker] [-k true/false] [--setup]")
```
Comment (rcannood) [inline src/main/scala/io/viash/cli/CLIConf.scala:217]:
```suggestion
      "viash test config.vsh.yaml [-p docker] [-k true/false] [--setup cachedbuild]")

File: src/main/scala/io/viash/cli/CLIConf.scala [hunk 3]
```diff
@@ -287,7 +294,13 @@ class CLIConf(arguments: Seq[String]) extends ScallopConf(arguments) {
       banner(
         "viash ns test",
         "Test a namespace containing many viash config files.",
-        "viash ns test [-n nmspc] [-s src] [-p docker] [--parallel] [--tsv file.tsv] [--append]")
+        "viash ns test [-n nmspc] [-s src] [-p docker] [--parallel] [--tsv file.tsv] [--setup] [--append]")
```
Comment (rcannood) [inline src/main/scala/io/viash/cli/CLIConf.scala:297]:
```suggestion
        "viash ns test [-n nmspc] [-s src] [-p docker] [--parallel] [--tsv file.tsv] [--setup cachedbuild] [--append]")

**Review by rcannood — CHANGES_REQUESTED** [review]
Minor suggestions :)

File: .github/workflows/sbt_test.yml (hunk 1 above)
Comment (DriesSchaumont) [reply .github/workflows/sbt_test.yml]: See https://github.com/viash-io/viash-actions/pull/15

File: .github/workflows/sbt_test.yml [hunk 4]
```diff
@@ -17,6 +17,9 @@ jobs:
     steps:
     - uses: actions/checkout@v3
 
+    - uses: viash-io/viash-actions/update-docker-engine@update-docker-engine
```
Comment (rcannood) [inline .github/workflows/sbt_test.yml:20]: Please use a release ^^

**Review by rcannood — CHANGES_REQUESTED** [review]

File: .github/workflows/sbt_test.yml (hunk 1 above)
Comment (rcannood) [reply .github/workflows/sbt_test.yml]: :+1:

File: .github/workflows/sbt_test.yml (hunk 4 above)
Comment (DriesSchaumont) [reply .github/workflows/sbt_test.yml]: Done!

**Pull request merged**